src/common/utils/http.ts

```diff
@@ -0,0 +1,59 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import { container } from "readium-desktop/main/di";
+import { RootState } from "readium-desktop/main/redux/states";
+import { AnyJson } from "readium-desktop/typings/json";
+import { Store } from "redux";
+import * as request from "request";
+import { promisify } from "util";
+
+/**
+ * @param url url of your GET request
+ * @param options request options
+ * @returns body of url response. 'String' type returned in many cases except for options.json = true
+ */
+export async function httpGet<T extends AnyJson = string>(url: string, options?: request.CoreOptions): Promise<T> {
```

Comment:
Hello @panaC you will be interested to know that the `request.CoreOptions` TypeScript type is incorrect when using stricter checking rules. I implemented a fix in my Pull Request:
https://github.com/readium/readium-desktop/pull/404/commits/79dbc9ed86646beced0421631ee215004d24324a